src/main/java/org/openrewrite/staticanalysis/RemoveUnusedParams.java

```diff
@@ -0,0 +1,158 @@
+package org.openrewrite.staticanalysis;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Repeat;
+import org.openrewrite.ScanningRecipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.NoMissingTypes;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.Statement;
+
+import java.time.Duration;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class RemoveUnusedParams extends ScanningRecipe<RemoveUnusedParams.Accumulator> {
+
+    static class Accumulator {
+        Set<String> overrideSignatures = new HashSet<>();
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Remove unused method parameters";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Removes parameters from methods that are declared but never used in the method body.";
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(5);
+    }
+
+    @Override
+    public Accumulator getInitialValue(ExecutionContext ctx) {
+        return new Accumulator();
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getScanner(Accumulator acc) {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+                J.MethodDeclaration m = super.visitMethodDeclaration(method, ctx);
+                for (J.Annotation ann : m.getLeadingAnnotations()) {
+                    if ("Override".equals(ann.getSimpleName())) {
+                        String key = m.getSimpleName() + "#" + m.getParameters().size();
+                        acc.overrideSignatures.add(key);
```

Comment:
do we get collision when SPOT?
```suggestion
                        acc.OVERRIDE_SIGNATURES.add(key);
```

Reply:
@timtebeek
Why did you marked this conversation as resolved ? It seems valid suggestion.

Reply:
@Pankraz76
Addressed as per the suggestion.